src/server/main_service.cc

```diff
@@ -2328,23 +2335,27 @@ void Service::OnClose(facade::ConnectionContext* cntx) {
   ConnectionState& conn_state = server_cntx->conn_state;
 
   if (conn_state.subscribe_info) {  // Clean-ups related to PUBSUB
+    optional<BlockingCounter> token;
+
     if (!conn_state.subscribe_info->channels.empty()) {
-      auto token = conn_state.subscribe_info->borrow_token;
+      token = conn_state.subscribe_info->borrow_token;
       server_cntx->UnsubscribeAll(false);
-
-      // Check that all borrowers finished processing.
-      // token is increased in channel_slice (the publisher side).
-      token.Wait();
     }
 
     if (conn_state.subscribe_info) {
       DCHECK(!conn_state.subscribe_info->patterns.empty());
-      auto token = conn_state.subscribe_info->borrow_token;
+      // DCHECK(!token || *token == conn_state.subscribe_info->borrow_token);
```

Comment:
remove the DCHECK ?

Reply:
I want to keep add in the future, but blocking counters are not comparable (should be by internal pointer, needs to be fixed in helio)